packages/react-form-renderer/src/form-renderer/render-form.js

```diff
@@ -15,18 +17,64 @@ FormFieldHideWrapper.defaultProps = {
   hideField: false
 };
 
-const FormConditionWrapper = ({ condition, children, field }) =>
-  condition ? (
-    <FormSpy subscription={{ values: true }}>
-      {({ values }) => (
-        <Condition condition={condition} values={values} field={field}>
-          {children}
-        </Condition>
-      )}
-    </FormSpy>
-  ) : (
-    children
+const ConditionTriggerWrapper = ({condition, values, children, field}) => (
+  <Condition condition={condition} values={values} field={field}>
+    {children}
+  </Condition>
+);
+
+ConditionTriggerWrapper.propTypes = {
+  condition: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
+  children: PropTypes.node.isRequired,
+  field: PropTypes.object,
+  values: PropTypes.object.isRequired,
+};
+
+const ConditionTriggerDetector = ({ values = {}, triggers = [], children, condition, field }) => {
+  const internalTriggers = [...triggers];
+  if (internalTriggers.length === 0) {
+    return (
+      <ConditionTriggerWrapper condition={condition} values={values} field={field}>
+        {children}
+      </ConditionTriggerWrapper>
+    );
+  }
+
+  const name = internalTriggers.shift();
+  return (
+    <FieldProvider skipRegistration name={name} subscription={{ value: true }} render={({input: {value}}) => (
```

Comment:
I would still use the FF field - we don't need any of our useFieldApi features (and so we don't have to use any `skipRegistration`)